packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -0,0 +1,421 @@
+  //   in the type byte (using _kNull, _kTrue, _kFalse)
+  // * Integers representable in 32 bits are encoded using 4 bytes big-endian,
+  //   two's complement representation.
+  // * Larger integers representable in 64 bits are encoded using 8 bytes
+  //   big-endian, two's complement representation.
+  // * Still larger integers are encoded using their hexadecimal string
+  //   representation. First the length of that is encoded in the expanding
+  //   format, then follows the UTF-8 representation of the hex string.
+  // * doubles are encoded using the IEEE 754 64-bit double-precision binary
+  //   format.
+  // * Strings are encoded using their UTF-8 representation. First the length
+  //   of that in bytes is encoded using the expanding format, then follows the
+  //   UTF-8 encoding itself.
+  // * Uint8Lists, Int32Lists, Int64Lists, and Float64Lists are encoded by first
+  //   encoding the list's element count in the expanding format, then the
+  //   smallest number of zero bytes needed to align the position in the full
+  //   message with a multiple of the number of bytes per element, then the
+  //   encoding of the list elements themselves, end-to-end with no additional
+  //   type information, using big-endian two's complement or IEEE 754 as
+  //   applicable.
+  // * Lists are encoded by first encoding their length in the expanding format,
+  //   then follows the recursive encoding of each element value, including the
+  //   type byte (Lists are assumed to be heterogeneous).
+  // * Maps are encoded by first encoding their length in the expanding format,
+  //   then follows the recursive encoding of each key/value pair, including the
+  //   type byte for both (Maps are assumed to be heterogeneous).
+  static const int _kNull = 0;
+  static const int _kTrue = 1;
+  static const int _kFalse = 2;
+  static const int _kInt32 = 3;
+  static const int _kInt64 = 4;
+  static const int _kLargeInt = 5;
+  static const int _kFloat64 = 6;
+  static const int _kString = 7;
+  static const int _kUint8List = 8;
+  static const int _kInt32List = 9;
+  static const int _kInt64List = 10;
+  static const int _kFloat64List = 11;
+  static const int _kList = 12;
+  static const int _kMap = 13;
+
+  const StandardMessageCodec();
+
+  @override
+  ByteData encodeMessage(dynamic message) {
+    if (message == null)
+      return null;
+    final WriteBuffer buffer = new WriteBuffer();
+    _writeValue(buffer, message);
+    return buffer.done();
+  }
+
+  @override
+  dynamic decodeMessage(ByteData message) {
+    if (message == null)
+      return null;
+    final ReadBuffer buffer = new ReadBuffer(message);
+    final dynamic result = _readValue(buffer);
+    if (buffer.hasRemaining)
+      throw new FormatException('Message corrupted');
+    return result;
+  }
+
+  static void _writeSize(WriteBuffer buffer, int value) {
+    assert(0 <= value && value < 0xffffffff);
```

Comment:
Expanding the format further was not the intention. I just had myself confused, that's all. Fix upcoming.